backend-project/small_eod/migration_v1/README.md

```diff
@@ -0,0 +1,44 @@
+
+###Bootstrap on dev:
+
+To run migrator locally:
+```shell script
+docker-compose up -d migration_db
+docker-compose exec migration_db mysql -ppass --execute 'CREATE DATABASE small_eod CHARACTER SET utf8;'
+docker-compose exec -T migration_db mysql small_eod -ppass < ./small_eod_demo-26-09-2020.sql
+docker-compose up -d --build
+docker-compose run --rm backend python manage.py migrate_v1
+```
+
+### Added unique constraints:
+
+I added some simple constraints that seemed missing from the data model:
+ - Channel.name
+ - Feature.name
+ - FeatureOption.('name', 'feature')
+ - DocumentType.name
+ - Tag.name
+
+### Migration assumptions:
+
+I had to make some assumptions about "what(v1)-goes-where(v2)" that may or may not be accurate:
+ - Dictionary.type is now Feature.name
+ - Dictionary.name is now FeatureOption.name
+ - LetterName model is now DocumentType
+ - Users responsible for case will want be notified
+ - Letter.identifier is now Letter.reference_number
+ 
+### Issues:
+ - v1 has no data for created_by and modified_by fields in general
+ - Case.comment field too short for source data
+ - Should Feature.name be translated?
+ - Should we subtract 1 while migrating Letter ordering?
+ - When migrating tags, should something be done with TagNamespace as well?
+ - There is no person model in v2 as it relies on users.User directly and
```

Comment:
We can add in small-eod v1 `Person` model relationship to `User`, fill it manually (9 rows) and use during migration.

Reply:
I don't see why add it in v1, but adding it in v2 may be a good idea if the table is soo small.

Adding following to the procedure would simplify code:
1. make sure all users exist in v2 instance by usual registration methods
2. fill in temporary `Person` table in v2
3. run migration, it will reference `Person` table for user data

Reply:
I would like add it in v1 to allow non-technical user fill that data in UI. 

We don't have any users in v2. We migrate all users from v1.

Since we are migrating users and adding the necessary information to the source data, we will be able to complete the migration in one go. Otherwise, we will have to have an intermediate stage of completing the user-person mapping.

Reply:
In that case, we can assume that `Person.user` always references valid user, correct?

Reply:
Yes

Reply:
Notes that we will have to create a new issue to add the appropriate field in `v1` (in Person relation to user) as PR to `master`. I leave it to you.

Reply:
I'm not sure if this went lost in the writing - the field exists, [but is is nullable](https://github.com/watchdogpolska/small_eod/blob/master/small_eod/cases/models.py#L123). As a result, Person records from demo dump have no actual relation to users, just nulls. In principle prod dump could have the same state, thus the problem was born. :)

I could make a PR that simply forces the Person.user column to be `NOT NULL`.

Reply:
Och, I miss that, because Django admin does not show that field as column in person list (see https://demo.small-eod.siecobywatelska.pl/admin/cases/person/ ). Fix available at #565 . In this case, I propose to force this field to be filled in at the validation stage before migration, and the business will guarantee that the data will be completed in all cases before migration.